imap_processing/cli.py

```diff
@@ -478,6 +480,14 @@ def do_processing(self, dependencies: list):
             datasets = ultra_l1a.ultra_l1a(dependencies[0])
             products = [write_cdf(dataset) for dataset in datasets]
             return products
+        elif self.data_level == "l1b":
+            data_dict = defaultdict(list)
```

Comment:
Below you are setting the field for data_dict to a dataset, you aren't appending to it, so I think this should just be a normal dictionary?
```suggestion
            data_dict = {}
```